src/groupeddataframe/splitapplycombine.jl

```diff
@@ -551,14 +630,16 @@ function _combine(gd::GroupedDataFrame,
     end
 
     idx_agg = Ref(NOTHING_IDX_AGG)
-    if length(gd) > 0 && any(x -> isagg(x, gd), cs_norm)
```

Comment:
I'm a bit lost by these changes in conditions, but I assume you know what you're doing and that it's tested. :-)

Reply:
This is very tricky. Several years ago when you introduced the fast aggregates we made an optimization that avoids calculation of `idx` if you use only fast aggregates. Additionally fast aggregates are only used if we have at least 1 group.

Now the functions we add `groupindices` and `proprow` are not fast aggregates, they also work correctly for 0-group data frames, and they do not require `idx` to be computed (but they can take advantage of it if it is already computed).

Finally - if we use fast aggregates we can upfront calculate the row lookup index that is called `idx_agg`.

How does it now work:
* if we only pass aggregates or new functions introduced in this PR we pre-compute `idx_agg` as it is used then. Before I did it only if `length(gd) > 0` but the cost of doing it also for `length(gd) == 0` is negligible and, in particular new functions take advantage of it also then.
* precompute the `gd.idx` always if `length(gd) == 0` (this is cheap) or we have at least one non-fast aggregate or new function case. It is safe.
* the change in assertion to `length(gd.idx) >= 0` is because in the future if the compiler is smart enough there was a risk that it would optimize out `gd.idx !== nothing` check and the new check is guaranteed not to be optimized out.